clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -229,7 +229,7 @@ public long connectionDelay(Node node, long now) {
      */
     @Override
     public boolean connectionFailed(Node node) {
-        return connectionStates.connectionState(node.idString()).equals(ConnectionState.DISCONNECTED);
+        return connectionStates.isDisconnected(node.idString());
```

Comment:
OK, maybe we can leave as is. @lindong28, can you address the other comment? Let's get this merged and we can do other improvements later.